projects/storefrontlib/shared/components/card/card.component.ts

```diff
@@ -101,6 +129,37 @@ export class CardComponent implements OnInit {
     this.editCard.emit(4);
   }
 
+  save(): void {
+    this.saveMode = false;
+    const saveCardEvent: SaveCardEvent = {
+      saveMode: this.saveMode,
+    };
+
+    Object.keys(this.saveForm.controls).forEach((control) => {
+      if (this.saveForm.get(control)?.touched) {
+        const value = this.saveForm.get(control)?.value;
+        if (value) {
+          saveCardEvent[control] = value;
+        }
```

Comment:
so an property will only be present in the event, if the value was changed. So any caller need to analysis the event if an property is existing. Why not just emitting the actual value for each control no matter whether it was changed or not?
why not have an object per control with attributes: changed and value. So the caller can rely on that the event will contain an object for each control and still has the data whether something was changed or not